testng-core/src/test/kotlin/org/testng/internal/ClassHelperTest.kt

```diff
@@ -0,0 +1,143 @@
+    }
+
+    @Test
+    fun testFindClassesInSameTest() {
+        runTest(
+            TestClassSample::class.java, 2, TestClassSample::class.java,
+            BabyPandaSample::class.java
+        )
+    }
+
+    @Test
+    fun testNoClassDefFoundError() {
+        val urlClassLoader = object : URLClassLoader(emptyArray<URL>()) {
+            override fun loadClass(className: String?): Class<*> {
+                throw NoClassDefFoundError()
+            }
+        }
+        ClassHelper.addClassLoader(urlClassLoader)
+        val fakeClassName = UUID.randomUUID().toString()
+        assertThat(ClassHelper.forName(fakeClassName))
+            .withFailMessage("The result should be null; no exception should be thrown.")
+            .isNull()
+    }
+
+    @Test(dataProvider = "data")
+    fun testWithDefaultMethodsBeingOverridden(
+        cls: Class<*>, expectedCount: Int, vararg expected: String
+    ) {
+        val methods = ClassHelper.getAvailableMethodsExcludingDefaults(cls)
+        assertThat(methods).hasSize(expectedCount)
+        methods.forEach {
+            val actual = it.declaringClass.name + "." + it.name
+            val isPresent = listOf(*expected).contains(actual)
+            assertThat(isPresent).isTrue()
+        }
+    }
+
+    @DataProvider(name = "data")
+    fun getTestData(): Array<Array<Any>> {
+        return arrayOf(
+            arrayOf(MickJaggerSample::class.java, 1, MickJaggerSample::class.java.name + ".dance"),
+            arrayOf(
+                JohnTravoltaMovesSample::class.java,
+                2,
+                JohnTravoltaMovesSample::class.java.name + ".walk",
+                AbstractMovesSample::class.java.name + ".dance"
+            ),
+            arrayOf(
+                BatmanSample::class.java,
+                3,
+                BatmanSample::class.java.name + ".fly",
+                BatmanSample::class.java.name + ".liftWeights",
+                BatmanSample::class.java.name + ".yellSlogan"
+            )
+        )
+    }
+
+    fun runTest(classToBeFound: Class<*>, expectedCount: Int, vararg classes: Class<*>) {
+        val xmlSuite = XmlSuite().apply {
+            name = "xml_suite"
+        }
+        listOf("test1", "test2", "test3").forEach {
+            newXmlTest(it, xmlSuite, *classes)
+        }
+        val xmlClasses = ClassHelper.findClassesInSameTest(classToBeFound, xmlSuite)
+        assertThat(xmlClasses).hasSize(expectedCount)
+    }
+
+    private fun newXmlTest(testName: String, xmlSuite: XmlSuite, vararg clazz: Class<*>) =
+        XmlTest(xmlSuite).apply {
+            name = testName
+            xmlClasses = mutableListOf<XmlClass?>().apply {
+                addAll(newXmlClass(*clazz))
+            }
+        }
+
+    private fun newXmlClass(vararg classes: Class<*>): List<XmlClass> {
+        return classes.map {
+            XmlClass(it)
+        }.toList()
+    }
+
+    private fun runTest(expected: List<String>, whichClass: Class<*>) {
```

Comment:
Same. The test is here.

Reply:
I didnt understand what you are hinting at here @juherr Can you please elaborate ?

Reply:
Ok. I got this. Fixed this.